FrontPage/AppDelegate.swift

```diff
@@ -50,4 +78,65 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     return false
   }
   
+  func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+    // time to register user with the "AeroGear UnifiedPush Server"
+    // perform registration of this device
+    do {
+      pushConfig.alias = "sample-app"
+      pushConfig.categories = ["testing", "sample"]
+      let device = Push.instance
+      
+      try device.register(deviceToken,
+                          pushConfig,
+                          success: {
+                            // successfully registered!
+                            print("successfully registered with UPS!")
+                            
+                            // send Notification for success_registered, will be handle by registered ViewController
+                            let notification = Notification(name: Notification.Name(rawValue: "success_registered"), object: nil)
+                            NotificationCenter.default.post(notification as Notification)
+      },
+                          failure: {(error: Error!) in
+                            print("Error Registering with UPS: \(error.localizedDescription)")
+                            
+                            let notification = Notification(name: Notification.Name(rawValue: "error_register"), object: nil)
+                            NotificationCenter.default.post(notification as Notification)
+      })
+    } catch {
+      print("Error while trying to register device:\n>>>>\n \(error)\n<<<<")
+    }
+  }
+  
+  func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable: Any], fetchCompletionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
```

Comment:
From reading the [docs for this method](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623013-application), I get a feeling that this method shouldn't really be used to just display messages that are received. My understanding is that this method should be used when the app is running in background mode and you want the app to fetch some data when a push message is received. It's very likely that the app is in background mode when this method is called, and thus the code below to show the message will never actually executed.

If the purpose of this code is to show messages that are received when the app is in foreground, I think the right method to implement is https://developer.apple.com/documentation/usernotifications/unusernotificationcenterdelegate/1649518-usernotificationcente.

I suggest taking a look at the docs and then choose the appropriate method to implement.